adm-zip.ts

```diff
@@ -0,0 +1,726 @@
+import { Constants, FileSystem, isWin, Errors, findFiles, writeFileTo, makeDir, writeFileToAsync } from "./util";
+const fs: typeof import('fs') = FileSystem.fileSystem();
+import pth from "path";
+import ZipEntry from "./zipEntry";
+import ZipFile from "./zipFile";
+import { ZipFileType } from "./types";
+
+const defaultOptions = {
+    // option "noSort" : if true it disables files sorting
+    noSort: false,
+    // read entries during load (initial loading may be slower)
+    readEntries: false,
+    // default method is none
+    method: Constants.NONE
+};
+
+function canonical(p: string) {
```

Comment:
Maybe these functions should be moved to utils. This way we can write tests for these functions.